tests/base.py

```diff
@@ -240,46 +244,56 @@ def max_bookmarks_by_stream(self, sync_records):
         """
         max_bookmarks = {}
         for stream, batch in sync_records.items():
-
-            upsert_messages = [m for m in batch.get('messages') if m['action'] == 'upsert']
-            stream_bookmark_key = self.expected_replication_keys().get(stream, set())
-            assert len(stream_bookmark_key) == 1  # There shouldn't be a compound replication key
-            stream_bookmark_key = stream_bookmark_key.pop()
-
-            bk_values = [message["data"].get(stream_bookmark_key) for message in upsert_messages]
-            max_bookmarks[stream] = {stream_bookmark_key: None}
-            for bk_value in bk_values:
-                if bk_value is None:
-                    continue
-
-                if max_bookmarks[stream][stream_bookmark_key] is None:
-                    max_bookmarks[stream][stream_bookmark_key] = bk_value
-
-                if bk_value > max_bookmarks[stream][stream_bookmark_key]:
-                    max_bookmarks[stream][stream_bookmark_key] = bk_value
+            # `transactions` is child stream of `orders` stream which is incremental.
+                # We are writing a separate bookmark for the child stream in which we are storing 
+                # the bookmark based on the parent's replication key.
+                # But, we are not using any fields from the child record for it.
+                # That's why the `transactions` stream does not have replication_key but still it is incremental.
```

Comment:
```suggestion
            # `transactions` is child stream of `orders` stream which is incremental.
            # We are writing a separate bookmark for the child stream in which we are storing 
            # the bookmark based on the parent's replication key.
            # But, we are not using any fields from the child record for it.
            # That's why the `transactions` stream does not have replication_key but still it is incremental.
```
Also, format this comment at every place in the code.

Reply:
Updated